src/s1reader/s1_burst_slc.py

```diff
@@ -423,6 +423,55 @@ def as_dict(self):
             self_as_dict[key] = val
         return self_as_dict
 
+    def bistatic_delay(self, xstep=1, ystep=1):
+        '''Computes the bistatic delay correction as
+        described in Gisinger et al, 2021
+
+        Parameters
+        ----------
+        xstep : int
+           spacing along x direction (range direction) in units of pixels
+
+        ystep : int
+           spacing along y direction (azimuth direction) in units of pixels
+
+
+        Returns
+        --------
+           LUT2D object of bistatic delay correction in seconds as a function
+           of the range and zimuth indices. This correction needs to be added
+           to the SLC tagged azimuth time to get the corrected azimuth times.
+        '''
+
+        pri = 1.0 / self.prf_raw_data
+        tau0 = self.rank * pri
+
+        mid_range = self.starting_range + 0.5 * self.width * self.range_pixel_spacing
+        tau_mid = mid_range * 2.0 / isce3.core.speed_of_light
+
+        nx = int(self.width / xstep)
+        ny = int(self.length / ystep)
+
+        x = np.arange(0, (nx+1)*xstep, xstep, dtype = int)
+        y = np.arange(0, (ny+1)*ystep, ystep, dtype = int)
+        x_mesh = np.meshgrid(x, y)[0]
+        slant_range = self.starting_range + x_mesh * self.range_pixel_spacing
+        tau = slant_range * 2.0 / isce3.core.speed_of_light
+
+        # the first term (tau_mid/2) is the bulk bistatic delay which was
+        # removed from the orginial azimuth time by the ESA IPF. Based on
+        # Gisinger et al, 2021, ESA IPF has used the mid of the second subswath
+        # to compute the bulk bistatic delay. However currently we have not
+        # been able to verify this from ESA documents. In this implementation
+        # we have used the range to the middle of the burst of interest to
+        # compute the bulk bistatic delay. The correction can be potentially
+        # biased by the amount of discrepancy between the ESA IPF bulk bistatic
+        # correction and our assumed middle of the burst of interest.
```

Comment:
Thanks @vbrancat . Yes that is very helpful. Actually that's exactly what I needed. 

Now we need to think about an implementation strategy on how to keep track of the near range to the mid subswath for each burst (not a very good choice they made really!).  For any burst in sub-swath 1 and 2 we have that "near range of mid subswath" as the "far range" and "near range" of the burst respectively. But for a burst in subswath 3 we are not keep tracking of the near range of subswath 2 and that may need some reorganizing of the reader. Alternatively we could assume for a burst in subswath 3, the near range to subswath 2 is : 

near_range_to_subswath_2 = near_range_burst_in_swath_3 - width*rangespacing

width being the width of the busrt in subswath 3 because we don't have a burst in subswath 2. If width is off even by 500 pixels, it would only lead to ~6 cm geolocation error in azimuth which is negligible in my opinion.

Reply:
@yunjunz  pointed out [ETAD ATBD](https://sentinels.copernicus.eu/documents/247904/4629150/Sentinel-1-ETAD-Algorithm-Technical-Baseline-Document.pdf) which equation 56 indicated middle of IW2. This is inconsistent with the Guide to S1 geocoding document. Still waiting on a confirmation from folks from ESA & DLR to see which one really the ESA IPF has used. 
Probably in the meantime it is useful to be prepared and keep track of iw1, iw2, iw3 near,mid,far ranges in the reader.

Reply:
The ETAD ATBD might probably be the latest version of what they are doing. However, in my mind, ETAD and IPF processor are two different things. I could not find any latest mention to bistatic delay correction in the latest Sentinel 1 product specification document